src/cloudWatchLogs/cloudWatchLogsUtils.ts

```diff
@@ -2,17 +2,38 @@
  * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
-
+import * as telemetry from '../shared/telemetry/telemetry'
 import * as vscode from 'vscode'
 import { CLOUDWATCH_LOGS_SCHEME } from '../shared/constants'
 import { fromExtensionManifest } from '../shared/settings'
-import { CloudWatchLogsGroupInfo } from './registry/logStreamRegistry'
+import { CloudWatchLogsData, CloudWatchLogsGroupInfo } from './registry/logStreamRegistry'
 import { CloudWatchLogsParameters } from './registry/logStreamRegistry'
 
 // URIs are the only vehicle for delivering information to a TextDocumentContentProvider.
 // The following functions are used to structure and destructure relevant information to/from a URI.
 // Colons are not valid characters in either the group name or stream name and will be used as separators.
 
+/**
+ * records a metric that the filter was successful IF a filter was actually applied i.e one of time or filterPattern were set.
+ * @param logData
+ * @param resourceType
+ */
+export function telemetryFilter(
+    logData: CloudWatchLogsData,
+    resourceType: telemetry.CloudWatchResourceType,
+    source: 'escapeHatch' | 'originalSearch'
+): void {
+    if (logData.parameters.startTime || logData.parameters.filterPattern) {
```

Comment:
Why do we skip if both are unset? Does some other metric get emitted?

Reply:
If neither is set then we aren't filtering the cloudwatchlogs, so I figured we shouldn't emit a metric that we are. The alternative would be to emit a metric with both of them as false, but that seemed weird to me. Is it still useful to see a filter metric if nothing is being filtered but a lot of the same code is running?

Reply:
Our telemetry metric names should reflect the action not necessarily the outcome. Someone removing existing filters is different from someone opening a fresh log group/stream without any filters despite having the same end result.

Reply:
I see. So a filter metric should still be emitted when they are both unset. Also, you highlight an important case that I missed: Undoing existing filters should definitely emit a metric.